compaction/time_window_compaction_strategy.cc

```diff
@@ -106,7 +106,23 @@ time_window_compaction_strategy_options::time_window_compaction_strategy_options
 time_window_compaction_strategy_options::time_window_compaction_strategy_options(const time_window_compaction_strategy_options&) = default;
 
 void time_window_compaction_strategy_options::validate(const std::map<sstring, sstring>& options, std::map<sstring, sstring>& unchecked_options) {
+    validate_compaction_window_unit(options);
+    unchecked_options.erase(COMPACTION_WINDOW_UNIT_KEY);
```

Comment:
this violates layering.
`validate_compaction_window_unit` do that since it's aware of `COMPACTION_WINDOW_UNIT_KEY`
and similarly the other validator helpers.